### PR TITLE
Add helper methods for getting metadata values

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc/issues/22728

I will add unit tests once we are happy with the API surface.

These helper methods follow what the Java Metadata collection does - https://github.com/grpc/grpc-java/blob/4a644cb8764afc946ff8325992d1bab1e0360ed3/api/src/main/java/io/grpc/Metadata.java#L239-L252 - and returns the last value.

I didn't go with an indexer because it feels weird in this situation. It would be get only (if set, what position would the entry be?), and it would either only get the last entry, or would throw an error if there are multiple. Neither feel like intuitive behavior for an indexer. The only type I can think of like this in .NET is NameValueCollection. Its indexer will concatinate strings together if there are multiple - https://docs.microsoft.com/en-us/dotnet/api/system.collections.specialized.namevaluecollection.item?view=netframework-4.8

tl;dr; I think a `Entry Get(string key)` method is a little less confusing.

@jtattermusch

### CODE REVIEW

File: src/csharp/Grpc.Core.Api/Metadata.cs [hunk 1]
```diff
@@ -75,7 +75,67 @@ internal Metadata Freeze()
...
+                {
+                    return entries[i];
+                }
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Gets the string value of the last metadata entry with the specified key. If there are no matching entries then <c>null</c> is returned.
```
Comment (jtattermusch) [inline src/csharp/Grpc.Core.Api/Metadata.cs:95]: nit: add a note what happens if the key exists, but the value is not of type string.

File: src/csharp/Grpc.Core.Api/Metadata.cs [hunk 2]
```diff
@@ -75,7 +75,67 @@ internal Metadata Freeze()
...
+        /// </summary>
+        public string GetValue(string key)
+        {
+            return Get(key)?.Value;
+        }
+
+        /// <summary>
+        /// Gets the bytes value of the last metadata entry with the specified key. If there are no matching entries then <c>null</c> is returned.
+        /// </summary>
+        public byte[] GetValueBytes(string key)
```
Comment (jtattermusch) [inline src/csharp/Grpc.Core.Api/Metadata.cs:105]: nit: add a note what happens if the key exists, but the value is not of type "bytes".

**Comment by JamesNK** [comment]
Done.

File: src/csharp/Grpc.Core.Api/Metadata.cs [hunk 3]
```diff
@@ -343,13 +349,14 @@ public byte[] ValueBytes
...
             /// <summary>
             /// Gets the string value of this metadata entry.
+            /// If the metadata entry is binary then an exception is thrown.
             /// </summary>
             public string Value
             {
                 get
                 {
                     GrpcPreconditions.CheckState(!IsBinary, "Cannot access string value of a binary metadata entry");
-                    return value ?? EncodingASCII.GetString(valueBytes);
```
Comment (jtattermusch) [inline src/csharp/Grpc.Core.Api/Metadata.cs:308]: why this change?
at the same time, we're keeping similar logic in ValueBytes, so I'd like to understand the reasoning behind it.

File: src/csharp/Grpc.Core.Tests/MetadataTest.cs [hunk 4]
```diff
@@ -242,6 +244,110 @@ public void FreezeMakesReadOnly()
...
+        {
+            var metadata = new Metadata
+            {
+                { "abc", "abc-value1" },
+                { "abc", "abc-value2" },
+                { "xyz", "xyz-value1" },
+                { "xyz-bin", Encoding.ASCII.GetBytes("xyz-value1") },
+            };
+
+            var abcEntry = metadata.GetValue("abc");
```
Comment (jtattermusch) [inline src/csharp/Grpc.Core.Tests/MetadataTest.cs:298]: nit: abcEntry should be abcValue? (copypaste from the Get() testcase)

Comment (jtattermusch) [reply src/csharp/Grpc.Core.Tests/MetadataTest.cs:298]: same for the tests below

File: src/csharp/Grpc.Core.Api/Metadata.cs (hunk 3 above)
Comment (JamesNK) [reply src/csharp/Grpc.Core.Api/Metadata.cs:308]: It is impossible to get here. If the IsBinary = true (which checks that value is null) then an error is thrown.

Correct me if I'm wrong.

Comment (jtattermusch) [reply src/csharp/Grpc.Core.Api/Metadata.cs:308]: I think you're right.
